Attempt to fix the Kafka flaky tests

cescoffier opened the pull request:
Today, we are starting three brokers side-by-side, which include three Kafka instances and three zookeeper instances. That may be too much and may lead to unexpected issues.
This commit split the tests into three different modules:

- kafka verifies the typical Kafka behavior (producer, consumer, codecs...)
- kafka-ssl verifies the connection with SSL
- kafka-sasl verifies the connection with SASL

Each of them starts a single broker with a specific configuration.

In the meantime, I've added a native test for SASL and divided the messaging ITs into 2 groups (one specific to Kafka, and another one for AMQP and others).

cescoffier committed:
Attempt to fix the Kafka CI issues

Today, we are starting three brokers side-by-side, which include three Kafka instances and three zookeeper instances. That may be too much and may lead to unexpected issues.
This commit split the tests into three different modules:

- kafka verifies the typical Kafka behavior (producer, consumer, codecs...)
- kafka-ssl verifies the connection with SSL
- kafka-sasl verifies the connection with SASL

Each of them starts a single broker with a specific configuration.

famod commented:
+💯 for tackling this! I won't be able to review this before this evening and I'm by no means a Kafka expert, so you don't have to wait for my review.

famod commented:
Btw, shall this fix #16371 XOR #16415 or both?

cescoffier commented:
Well... actually it does not fixes the issue. I will revisit this.

cescoffier committed:
Rewrote Kafka SSL tests

* the previous code was actually not configured SSL consistently
* switch to test container (Strimzi containers)
* configure the native test to actually use SSL

cescoffier committed:
Rewrote Kafka SASL tests

- Switch to Test Containers (using Strimzi containers)
- Make sure SASL is configured consistently

cescoffier committed:
Switch from Debezium to Test Containers to start the Kafka brokers

Debezium requires Kafka and Scala which lead to issues when updating these libraries (alignment)

gastaldi reviewed (approved):
LGTM

cescoffier commented:
@quarkusbot you're drunk... the build succeeded...

@gsmet ^^

gsmet commented:
It's not the bot that has an issue: the workflow was reported as a failure: https://github.com/quarkusio/quarkus/actions/runs/744900055

Apparently, we have some issue in the workflow file with the native test json

gsmet commented:
There was an issue in the native tests JSON file. So thanks to the drunk bot for catching it :).

famod commented on integration-tests/kafka-sasl/src/test/java/io/quarkus/it/kafka/KafkaSASLTestResource.java:
```diff
@@ -0,0 +1,124 @@
+    public Map<String, String> start() {
+        kafka.start();
+        // Used by the test
+        System.setProperty("bootstrap.servers", kafka.getBootstrapServers());
```
Shouldn't the test be able to inject `kafka.bootstrap.servers` (which is added to the map further down) via `@ConfigProperty`? 🤔

cescoffier replied:
That's actually on purpose to distinguish the value retrieve in the test, and the once injected in the application. You can't inject the configuration value in the (native) test, so I use a system property.

famod replied:
Ah, of course! Thanks.

famod reviewed (approved):
Phew, a lot to take in and as I said, not a Kafka (or Strimizi, funny name) expert, but LGTM!

I only added a small question.

I haven't had a look at the build logs whether the tests are actually running but I assume you did. 🙂
